vmdb/app/models/ems_refresh/parsers/foreman.rb

```diff
@@ -0,0 +1,88 @@
+module EmsRefresh::Parsers
+  class Foreman
+    def self.ems_inv_to_hashes(inv)
+      new.ems_inv_to_hashes(inv)
+    end
+
+    def ems_inv_to_hashes(inv)
+      result = {}
+      ids = {}
+      result[:customization_scripts] = customization_scripts_inv_to_hash(inv[:media], inv[:ptables])
+      add_ids(ids, result[:customization_scripts])
+      result[:operating_system_flavors] = operating_system_flavors_inv_to_hashes(inv[:operating_systems], ids)
+      add_ids(ids, result[:operating_system_flavors])
+      result[:configuration_profiles] = configuration_profile_inv_to_hashes(inv[:hostgroups], ids)
+      add_ids(ids, result[:configuration_profiles])
+      result[:configured_systems] = configured_system_inv_to_hashes(inv[:hosts], ids)
+      result
+    end
+
+    def media_inv_to_hashes(media)
+      media.map do |m|
+        {
+          "manager_ref"  => "medium:#{m["id"]}",
+          "type"         => "CustomizationScriptMedium",
+          "name"         => m["name"]
+        }
+      end
+    end
+
+    def ptables_inv_to_hashes(ptables)
+      ptables.map do |m|
+        {
+          "manager_ref"  => "ptables:#{m["id"]}",
+          "type"         => "CustomizationScriptPtable",
+          "name"         => m["name"]
+        }
+      end
+    end
+
+    def customization_scripts_inv_to_hash(media, ptables)
+      media_inv_to_hashes(media) + ptables_inv_to_hashes(ptables)
+    end
+
+    def operating_system_flavors_inv_to_hashes(flavors_inv, ids)
+      # convert to a hash
+      flavors_inv.map do |os|
+        {
+          "manager_ref"  => "operating_system:#{os["id"]}",
+          "name"         => os["fullname"],
+          "description"  => os["description"],
+          "customization_scripts" => (os["media"] + os["ptables"]).each { |script| ids[script] },
+        }
+      end
+    end
+
+    def configuration_profile_inv_to_hashes(recs, ids)
+      recs.map do |profile|
+        {
+          "manager_ref"                 => "hostgroup:#{profile["id"]}",
+          "name"                        => profile["name"],
+          "description"                 => profile["title"],
+          "operating_system_flavor"     => ids[profile["operatingsystem_id"]],
+          "customization_script_ptable" => ids[profile["ptable_id"]],
+          "customization_script_medium" => ids[profile["medium_id"]],
+        }
+      end
+    end
+
+    def configured_system_inv_to_hashes(recs, ids)
+      recs.map do |cs|
+        {
+          "manager_ref"             => "host:#{cs["id"]}",
+          "hostname"                => cs["name"],
+          "configuration_profile"   => ids[cs["hostgroup_id"]],
+          "operating_system_flavor" => cs["operatingsystem_id"],
+        }
+      end
+    end
+
+    private
+
+    def add_ids(target, recs, key = "manager_ref")
+      recs.each_with_object(target) do |rec, dest|
+        dest[rec[key]] = rec
+      end
```

Comment:
I'm not questioning the method...I'm questioning the implementation...my example above does the same thing with less code and without calling each_with_object.